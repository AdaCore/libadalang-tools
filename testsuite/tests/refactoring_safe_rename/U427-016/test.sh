safe_rename -P default.gpr -S main.adb -L 8 -R 4 -N Foo --algorithm analyse_ast
safe_rename -P default.gpr -S main.adb -L 8 -R 4 -N Bar --algorithm analyse_ast
safe_rename -P default.gpr -S main.adb -L 8 -R 4 -N Cursor --algorithm analyse_ast
safe_rename -P default.gpr -S main.adb -L 8 -R 4 -N List --algorithm analyse_ast
safe_rename -P default.gpr -S main.adb -L 8 -R 4 -N File_Type --algorithm analyse_ast
