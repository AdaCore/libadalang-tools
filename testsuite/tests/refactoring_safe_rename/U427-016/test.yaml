description: |
    This test checks if the safe rename tool correctly handles different kinds
    of packages used on a Use_Package_Clause. The kinds of packages tested by
    this test are: Package_Decl, Package_Renaming_Decl and
    Generic_Package_Instantiation.
    A use clause with multiple packages separated by a comma is also tested.

driver: shell_script
