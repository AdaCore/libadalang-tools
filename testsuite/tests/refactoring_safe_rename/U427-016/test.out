# Renaming node main.adb:8:4: 
References:
Unit:main.adb
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 4
End_Line: 8
End_Column: 7
Line: 25
Source Location Range:
Start_Line: 25
Start_Column: 14
End_Line: 25
End_Column: 17
Problems:
Renaming Baz to Foo hides <Id "Foo" d.ads:3:4-3:7>

# Renaming node main.adb:8:4: 
References:
Unit:main.adb
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 4
End_Line: 8
End_Column: 7
Line: 25
Source Location Range:
Start_Line: 25
Start_Column: 14
End_Line: 25
End_Column: 17
Problems:
Renaming Baz to Bar hides <Id "Bar" f.ads:3:4-3:7>

# Renaming node main.adb:8:4: 
References:
Unit:main.adb
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 4
End_Line: 8
End_Column: 7
Line: 25
Source Location Range:
Start_Line: 25
Start_Column: 14
End_Line: 25
End_Column: 17
Problems:
Renaming Baz to Cursor hides <| Id "Cursor" a-convec.ads:107:9-107:15 [h.ads:3:1] |>

# Renaming node main.adb:8:4: 
References:
Unit:main.adb
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 4
End_Line: 8
End_Column: 7
Line: 25
Source Location Range:
Start_Line: 25
Start_Column: 14
End_Line: 25
End_Column: 17
Problems:
Renaming Baz to List hides <| Id "List" a-cdlili.ads:54:9-54:13 [i.ads:3:1] |>

# Renaming node main.adb:8:4: 
References:
Unit:main.adb
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 4
End_Line: 8
End_Column: 7
Line: 25
Source Location Range:
Start_Line: 25
Start_Column: 14
End_Line: 25
End_Column: 17
Problems:
Renaming Baz to File_Type hides <Id "File_Type" a-textio.ads:65:9-65:18>

